adapters/metax/metax.go

```diff
@@ -0,0 +1,196 @@
+package metax
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/prebid/prebid-server/v2/util/ptrutil"
+)
+
+const SupportedCurrency = "USD"
+
+type adapter struct {
+	template *template.Template
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	// split impressions
+	reqDatas := make([]*adapters.RequestData, 0, len(request.Imp))
+	for i := range request.Imp {
+		imp := &request.Imp[i]
+		impCopy := *imp
+		requestCopy := *request
```

Comment:
U dont have to split impressions like that. U can just use. In Go (newest vers) You can iterate using range
	reqCopy := *request
	for _, imp := range request.Imp {
		reqCopy.Imp = []openrtb2.Imp{imp}

Reply:
as I know that moving to for _, imp := range request.Imp will copy the Imp struct and increase the memory usage
so I loop the slice by index

Reply:
I just understand what you told, it's updated now